tests/integration/helm/install_test.go

```diff
@@ -0,0 +1,218 @@
+			if err != nil {
+				t.Fatal(err)
+			}
+			overrideValuesStr := `
+global:
+  hub: %s
+  tag: %s
+  jwtPolicy: first-party-jwt
+`
+			overrideValues := fmt.Sprintf(overrideValuesStr, s.Hub, s.Tag)
+			overrideValuesFile := filepath.Join(workDir, "values.yaml")
+			if err := ioutil.WriteFile(overrideValuesFile, []byte(overrideValues), os.ModePerm); err != nil {
+				t.Fatalf("failed to write iop cr file: %v", err)
+			}
+			installIstio(t, ctx, cs, h, overrideValuesFile)
+
+			verifyInstallation(t, ctx, cs)
+
+			t.Cleanup(func() {
+				deleteIstio(t, cs, h)
+			})
+		})
+}
+
+// installIstio install Istio using Helm charts with the provided
+// override values file and fails the tests on any failures.
+func installIstio(t *testing.T, ctx resource.Context, cs resource.Cluster,
+	h *helm.Helm, overrideValuesFile string) {
+	if _, err := cs.CoreV1().Namespaces().Create(context.TODO(), &kubeApiCore.Namespace{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name: IstioNamespace,
+		},
+	}, kubeApiMeta.CreateOptions{}); err != nil {
+		_, err := cs.CoreV1().Namespaces().Get(context.TODO(), IstioNamespace, kubeApiMeta.GetOptions{})
+		if err == nil {
+			log.Info("istio namespace already exist")
+		} else {
+			t.Errorf("failed to create istio namespace: %v", err)
+		}
+	}
+
+	// Install base chart
+	err := h.InstallChart(BaseReleaseName, BaseChart,
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", BaseChart)
+	}
+
+	// Install discovery chart
+	err = h.InstallChart(IstiodReleaseName, filepath.Join(ControlChartsDir, DiscoveryChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", DiscoveryChart)
+	}
+
+	// Install ingress gateway chart
+	err = h.InstallChart(IngressReleaseName, filepath.Join(GatewayChartsDir, IngressGatewayChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", IngressGatewayChart)
+	}
+
+	// Install egress gateway chart
+	err = h.InstallChart(EgressReleaseName, filepath.Join(GatewayChartsDir, EgressGatewayChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", EgressGatewayChart)
+	}
+}
+
+// deleteIstio deletes installed Istio Helm charts and resources
+func deleteIstio(t *testing.T, cs resource.Cluster, h *helm.Helm) {
+	scopes.Framework.Infof("cleaning up resources")
+	if err := h.DeleteChart(IngressReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", IngressReleaseName)
+	}
+	if err := h.DeleteChart(EgressReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", EgressReleaseName)
+	}
+	if err := h.DeleteChart(BaseReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", BaseReleaseName)
+	}
+	if err := cs.CoreV1().Namespaces().Delete(context.TODO(), IstioNamespace, kubeApiMeta.DeleteOptions{}); err != nil {
```

Comment:
Running tests in succession is causing intermittent failures due to the namespace terminating while the new Helm install happens. Is there a util to ensure a namespace is completely gone before proceeding with the next step?